include/linker/common-rom.ld

```diff
@@ -131,6 +131,15 @@
 
 	Z_ITERABLE_SECTION_ROM(k_p4wq_initparam, 4)
 
+#if defined(CONFIG_FLASH_MAP)
+	SECTION_DATA_PROLOGUE(_flash_map,,SUBALIGN(4))
+	{
+		__flash_map_list_start = .;
```

Comment:
The commit log here says what is being done, but not why, so I'm not sure what this is for. The pull request description doesn't mention the linker either.

What is the goal? Why do we need a separate linker section? It's not clear why this is the first patch in the series.

Reply:
The separate linker section is used for ordering the areas so that code, outside of zephyr, that still uses id could be picking them by id (index). This also required change in gen_defines.py because only ID for "okay" partitions are now expected, so that they would be continuous.